imagetest/test_suites/metadata/setup.go

```diff
@@ -12,40 +12,189 @@ var Name = "metadata"
 
 const (
 	daemonScriptTemplate = `#!/bin/bash
-
-nohup sleep 3600 > /dev/null 2>&1 < /dev/null &
-echo $! > %s
-`
+	nohup sleep 3600 > /dev/null 2>&1 < /dev/null &
+	echo $! > %s
+	`
 	daemonOutputPath = "/daemon_out.txt"
 
-	startupScriptTemplate = `#!/bin/bash
-echo "%s" > %s`
-	startupOutputPath  = "/startup_out.txt"
-	startupContent     = "The startup script worked."
-	shutdownScriptTime = `#!/bin/bash
-
-while [[ 1 ]]; do
-  date +%s >> /shutdown.txt
-  sync
-  sleep 1
-done`
-	shutdownScriptTemplate = `#!/bin/bash
-echo "%s" > %s`
-	shutdownOutputPath = "/shutdown_out.txt"
-	shutdownContent    = "The shutdown script worked."
+	bashScriptTemplate = `#!/bin/bash
+	echo "%s" > %s`
+	windowsScriptTemplate = `echo "%s" > %s
+	ping -n 2 127.0.0.1 >NUL`
+	powershellScriptTemplate = `Set-Content -Value "%s" -Path %s
+	Start-Sleep 1`
+	startupContent  = "The startup script worked."
+	shutdownContent = "The shutdown script worked."
+
+	timeOutputPath        = "/shutdown.txt"
+	windowsTimeOutputPath = "C:\\shutdown.txt"
+
+	bashTimeScriptTemplate = `#!/bin/bash
```

Comment:
this naming is more generic than 'shutdownScriptTime', and not clear why you've made it a template since it's only used once

Reply:
The template isn't used more than once, but the path variables are used more than once.